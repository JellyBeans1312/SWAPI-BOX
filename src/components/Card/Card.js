import React from 'react';

const Card = ({ name, species, homeworld}) => {
  return (
    <article>
      <h3>{name}</h3>
      <button>Favorite me!</button>
      <p>Homeworld: Tatooine</p>
      <p>Species: {species}</p>
      <p>Population: 100000</p>
    </article>
  )
}

export default Card;