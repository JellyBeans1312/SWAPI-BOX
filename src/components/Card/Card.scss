.card {
  box-sizing: border-box;
  margin: 10px;
  height: 220px;
  width: 250px;
  background: rgba(155, 155, 155, .2);
  padding-left: 10px;
  color: white;
}