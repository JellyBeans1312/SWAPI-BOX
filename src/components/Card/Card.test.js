import Card from './Card';
import { shallow } from 'enzyme';
import React from 'react'

describe('Card', () => {
  let wrapper;
  let fakeData = [
    'hello',
    'there',
    'person',
    'of',
    'interest'
  ];
  const mockFavorite = jest.fn()
  beforeEach(() => {
    wrapper =  shallow(
      <Card 
        datum={fakeData}
        favoriteStatus={mockFavorite}
      />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot() 
  });

  it('should call favorite method on click', () => {
    wrapper.find('img').simulate('click')
    expect(mockFavorite).toHaveBeenCalled()
  })
});
