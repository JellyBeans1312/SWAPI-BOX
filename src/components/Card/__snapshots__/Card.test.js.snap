// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<article
  className="card"
>
  <div
    class="card-title"
  >
    <h3>
      hello
    </h3>
    <img
      alt=""
      class="star-card-button"
      onClick={[Function]}
      src="star.svg"
    />
  </div>
  <div
    className="card_content"
  >
    <p>
      there
    </p>
    <p>
      person
    </p>
    <p>
      of
    </p>
    <p>
      interest
    </p>
  </div>
</article>
`;
